Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1355,9 +1355,41 @@ static LayoutUnit computeSubgridMarginBorderPadding(const RenderGrid* outermost,
 
 void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track, unsigned trackIndex, GridIterator& iterator, Vector<GridItemWithSpan>& itemsSortedByIncreasingSpan, Vector<GridItemWithSpan>& itemsCrossingFlexibleTracks, SingleThreadWeakHashSet<RenderBox>& itemsSet, LayoutUnit currentAccumulatedMbp)
 {
-    while (auto* gridItem = iterator.nextGridItem()) {
+    // This is for masonry items and helps track if we visited the item or not when going through this specific track.
+    HashMap<RenderBox*, bool> visitedGridItems;
```

Comment:
Why not just a (Weak)HashSet?

Reply:
Good point; updating to `SingleThreadWeakHashSet`.